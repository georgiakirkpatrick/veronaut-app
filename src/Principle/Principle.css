.Principle {
    padding: 5vh 0;
    line-height: 16px;
}

.Principle > h3 {
    font: 500 18px Poppins;
    letter-spacing: 1px;
    padding: 3vh 0;
    color: #041516;
}

.Principle > p {
    font: 300 16px/20px Poppins;
    line-height: 20px;
    letter-spacing: 1px;
    color: #041516;
}

.Principle > svg {
    color: #041516;
}

@media screen and (min-width: 768px) {
    .Principle {
        padding: 2vh 0;
        line-height: 16px;
    }
}