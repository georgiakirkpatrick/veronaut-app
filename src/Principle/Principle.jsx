import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import '../FontAwesomeIcons/FontAwesomeIcons';
import React from 'react';
import './Principle.css';

const Principle = props => {
  const {
    title,
    description,
    symbol
  } = props

  return (
    <li className='Principle'>
      <FontAwesomeIcon icon={symbol} size='lg' />
      <h3>{title}</h3>
      <p>{description}</p>
    </ li>
  )
}

Principle.defaultProps = {
  title: '',
  description: '',
  symbol: ''
}

export default Principle;