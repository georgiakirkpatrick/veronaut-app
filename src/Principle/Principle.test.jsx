import React from 'react';
import { render, screen } from '@testing-library/react';
import Principle from './Principle';
import { describe, it, expect } from 'vitest';

const principleDescription = 'We value versatile styles that bring out our best selves in the office and at happy hour.  That hold up on our bike commutes and school pick-ups.  Our ideal clothing is bra-friendly, easy care, and has pockets.'
const principleTitle = 'Good design'

describe ('Principle', () => {
    it ('renders without crashing', () => {
        render(<Principle />)
        screen.debug()
    })

    it('renders the UI as expected', () => {
        const tree = render(<Principle principleTitle={principleTitle} principleDescription={principleDescription} />)
        expect(tree).toMatchSnapshot()    
    })
})