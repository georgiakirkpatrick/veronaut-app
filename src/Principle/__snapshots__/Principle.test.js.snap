// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Principle renders the UI as expected 1`] = `
<li
  className="Principle"
>
  <h3>
    
  </h3>
  <p>
    
  </p>
</li>
`;
